src/main/scala/com/cognite/sdk/scala/common/dataTypes.scala

```diff
@@ -260,6 +267,37 @@ object NonNullableSetter {
   ): Encoder[NonNullableSetter[T]] = new Encoder[NonNullableSetter[T]] {
     final def apply(a: NonNullableSetter[T]): Json = a match {
       case SetValue(value) => Json.obj(("set", encodeT.apply(value)))
+      case UpdateArray(_, _) => ???
```

Comment:
This would be my suggestion, it's annoying that we need the "this should never happen" exception... but it really never should, as long as the compiler picks the most specific implicit conversion.

```scala
  implicit def encodeNonNullableSetterSeq[T](
      implicit encodeT: Encoder[T]
  ): Encoder[NonNullableSetter[Seq[T]]] = new Encoder[NonNullableSetter[Seq[T]]] {
    private val encodeSeqT = Encoder.encodeSeq(encodeT)

    final def apply(a: NonNullableSetter[Seq[T]]): Json = a match {
      case SetValue(value) =>
        Json.obj("set" -> encodeSeqT(value))
      case UpdateArray(add: Seq[T] @unchecked, remove: Seq[T] @unchecked) =>
        Json.obj("add" -> encodeSeqT(add), "remove" -> encodeSeqT(remove))
    }
  }

  private val encodeMapStringString =
    Encoder.encodeMap[String, String](KeyEncoder.encodeKeyString, Encoder.encodeString)
  private val encodeSeqString = Encoder.encodeSeq[String]
  implicit def encodeNonNullableSetterMapStringString
      : Encoder[NonNullableSetter[Map[String, String]]] =
    new Encoder[NonNullableSetter[Map[String, String]]] {
      final def apply(a: NonNullableSetter[Map[String, String]]): Json = a match {
        case SetValue(value) =>
          Json.obj("set" -> encodeMapStringString(value))
        case UpdateMap(add, remove) =>
          Json.obj("add" -> encodeMapStringString(add), "remove" -> encodeSeqString(remove))
      }
    }

  implicit def encodeNonNullableSetter[T](
      implicit encodeT: Encoder[T]
  ): Encoder[NonNullableSetter[T]] = new Encoder[NonNullableSetter[T]] {
    final def apply(a: NonNullableSetter[T]): Json = a match {
      case SetValue(value) =>
        Json.obj("set" -> encodeT(value))
      case _ =>
        throw new RuntimeException("Invalid NonNullableSetter. This should never happen.")
    }
  }
```